api/Metrics/LabelableMetric.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Metrics;
+
+interface LabelableMetric extends Metric
+{
+    /**
+     * Get $labels
+     *
+     * @return  array
+     */
+    public function getLabels(): array;
+
+    /**
+     * Set $labels
+     *
+     * @param  array  $labels
+     * @return  self
+     */
+    public function setLabels(array $labels);
```

Comment:
I don't like unstructured arrays like this. We should either:
- Make interfaces that explicitly specify all type info (which is a bit tedious)
- Or add structure to the arrays in the doc comments.

I have a preference for the first one, as not everyone will use static analysis tools and those types will all be enforced at runtime, but may accept the latter.

Reply:
Done.

For now I've chosen the second option but added todo for a later refactoring.
I also added this check https://github.com/open-telemetry/opentelemetry-php/blob/56caa169c9f032695c491f15f00d3ce323128247/sdk/Metrics/HasLabels.php#L29 to make sure `setLabels` receives only arrays of strings